File: 2685-count-the-number-of-complete-components/2685-count-the-number-of-complete-components.java
class Solution {
     static int e=0,v=0;
    public static int countCompleteComponents(int n, int[][] edges) {
        List<List<Integer>> adj = new ArrayList<>();
        for (int i = 0; i < n; i++) adj.add(new ArrayList<>());
        for (int[] edge: edges) {
            adj.get(edge[0]).add(edge[1]);
            adj.get(edge[1]).add(edge[0]);
        }
        boolean[] visited = new boolean[n];
        int c = 0;
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(visited, i, adj);
                int Vertices_irunthu_yethana_lines_pogum_1vertice_la_irunthu_2lines_poganumMUST = v*(v-1) /2 ; // appidi pochuna 
                if(Vertices_irunthu_yethana_lines_pogum_1vertice_la_irunthu_2lines_poganumMUST *2==e)c++;
            }
            e=0;
            v=0;
        }
        return c;
    }

    public static void dfs(boolean[] visited, int from, List<List<Integer>> adj) {
        visited[from] = true;
        v++;
        for (int to : adj.get(from)) {
            e++;
            if (!visited[to]) {
                dfs(visited, to, adj);
            }
        }
    }
}

File: 2685-count-the-number-of-complete-components/2685-count-the-number-of-complete-components.cpp
class Solution {
private:
    void dfs(int node, vector<int>&vis, vector<vector<int>>&adj,int &nodes,int &edge){
       vis[node] = 1;
       nodes++;                     // Ek node visit hua
       edge += adj[node].size();   // Is node ki degree add karo
        for(auto it:adj[node]){
            if(!vis[it]){
                dfs(it,vis,adj,nodes,edge);
            }
        }
    }
public:
    int countCompleteComponents(int n, vector<vector<int>>& edges) {
         // convert into adj list
        vector<vector<int>>adj(n);
        for(auto &it:edges){
            int u=it[0];
            int v=it[1];
            adj[u].push_back(v);
            adj[v].push_back(u);
        }

        // use dfs 
        vector<int>vis(n,0);
        int cnt=0;
        
        for(int i=0; i<n; i++){
            if( !vis[i]){
                int edge=0;
                int nodes=0;
                dfs(i,vis,adj,nodes,edge);
                edge /=2;
                if(edge==nodes*(nodes-1)/2)
                  cnt++;
            }
        }
        return cnt;
        
    }
};